dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -931,18 +932,15 @@ bool DAGExpressionAnalyzer::buildExtraCastsAfterTS(
             const auto fsp = table_scan_columns[i].decimal < 0 ? 0 : table_scan_columns[i].decimal;
             tipb::Expr fsp_expr = constructInt64LiteralTiExpr(fsp);
             fsp_col = getActions(fsp_expr, actions);
-            String casted_name = appendDurationCast(fsp_col, source_columns[i].name, dur_func_name, actions);
-            source_columns[i].name = casted_name;
+            casted_name = appendDurationCast(fsp_col, source_columns[i].name, dur_func_name, actions);
             source_columns[i].type = actions->getSampleBlock().getByName(casted_name).type;
```

Comment:
Should not change the type of source_columns here

Reply:
should change. `a, b, c, cast_c` -> `a, b, c`, here, `cast_c` is renamed to `c`, so `source_columns[2].type` should equal to `cast_c`'s type